File: daily/python/Dec-11-Element-appearing-more-than-23p-in-sorted-array.py
class Solution:
    def findSpecialInteger(self, arr: List[int]) -> int:
        leng = len(arr)
        count = 0
        prev = arr[0]
        max = 0
        for i in range(leng):
            if (arr[i] == prev):
                count += 1
            else:
                count = 1
                prev = arr[i]

            if count > max:
                max = count
            if max > (leng // 4):
                return prev

        return prev

File: daily/cpp/Dec-11-Element-appearing-more-than-23p-in-sorted-array.cpp
using namespace std;

class Solution {
 public:
  int findSpecialInteger(vector<int>& arr) {
    int len = arr.size();
    int count = 0;
    int prev = arr[0];
    int max = 0;
    for (int i = 0; i < len; ++i) {
      if (arr[i] == prev) {
        count++;
      } else {
        count = 1;
        prev = arr[i];
      }
      if (count > max) {
        max = count;
      }
      if (max > len / 4) {
        return prev;
      }
    }
    return prev;
  }
};
